WorkflowUI/Sources/Screen/AdaptedEnvironmentScreen.swift

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(UIKit)
+
+import Foundation
+import ViewEnvironment
+
+/// Wraps a `Screen` tree with a modified `ViewEnvironment`.
+///
+/// By specifying environmental values with this `Screen`, all child screens nested
+/// will inherit those values automatically. Values can be changed
+/// anywhere in a sub-tree by inserting another `AdaptedEnvironmentScreen`.
+///
+/// ```swift
+/// MyScreen(...)
+///     .adaptedEnvironment(keyPath: \.myValue, to: newValue)
+/// ```
+///
+public struct AdaptedEnvironmentScreen<Content: Screen>: Screen {
```

Comment:
Looking at https://github.com/square/workflow-swift/pull/106 from @bencochran, I think I should likely move this conformance to an extension.

Reply:
https://github.com/square/workflow-swift/pull/198